content/en/blog/2020/show-source-ip/index.md

```diff
@@ -0,0 +1,336 @@
+---
+title: How to enable proxy protocol on AWS NLB and Istio ingress gateway
+subtitle: How to enable proxy protocol on AWS NLB and Istio ingress gateway
+description: How to enable proxy protocol on AWS NLB and Istio ingress gateway.
+publishdate: 2020-11-18
+attribution: "Xinhui Li (Salesforce) "
+keywords: [trafficManagement,protocol extending]
+---
+
+This blog presents my latest experience about how to configure/enable proxy protocol with stack of AWS NLB and Istio Ingress gateway. The [Proxy Protocol](https://www.haproxy.com/blog/haproxy/proxy-protocol/) was designed to chain proxies / reverse-proxies without losing the client information with the benefits from protocol agnostic, no need change infrastructure, no negative impact from `NATing` firewalls and good scalability.  To get client IP easy to read, it is also enabled `X-Forwarded-For` of HTTP header in the deployment. In this blog, traffic management of Istio ingress are shown with httpbin service on 80 and 443 respectively to demonstrate the enabling result. Below comes a part of the protocol describing about the difference of proxy protocol v1 and v2.  In our use case, both v1 and v2 works for the purpose but AWS NLB today only support v2 shown by image below. So, proxy protocol v2 is referred in the rest of this blog by default.
+
+{{< text yaml >}}
+A receiver may be configured to support both version 1 and version 2 of the
+protocol. Identifying the protocol version is easy :
+    - if the incoming byte count is 16 or above and the first 13 bytes match
+      the protocol signature block followed by the protocol version 2 :
+           \x0D\x0A\x0D\x0A\x00\x0D\x0A\x51\x55\x49\x54\x0A\x02
+    - otherwise, if the incoming byte count is 8 or above, and the 5 first
+      characters match the US-ASCII representation of "PROXY" then the protocol
+      must be parsed as version 1 :
+           \x50\x52\x4F\x58\x59
+    - otherwise the protocol is not covered by this specification and the
+      connection must be dropped.
+{{< /text >}}
```

Comment:
```suggestion
{{< tip >}}
A receiver may be configured to support both version 1 and version 2 of the
protocol. Identifying the protocol version is easy:

- if the incoming byte count is 16 or more and the first 13 bytes match the protocol signature block
   `\x0D\x0A\x0D\x0A\x00\x0D\x0A\x51\x55\x49\x54\x0A\x02`, the protocol is version 2. 
           
- otherwise, if the incoming byte count is 8 or more, and the 5 first characters match the `US-ASCII`
    representation of "PROXY" (`\x50\x52\x4F\x58\x59`), then the protocol must be parsed as version 1.

- otherwise the protocol is not covered by this specification and the connection must be dropped.
{{< /tip >}}
```